### ISSUE TITLE
"Error: No support for nested multi bulk replies with depth > 7" error

### ISSUE THREAD

**Issue by abrookins** [RediSearch/RediSearch#1722]
Hey, folks! I'm trying to get scores for a query I'm testing, and when I do, redis-cli crashes with the following error:

```
127.0.0.1:6379> FT.SEARCH books-idx "((@categories:{History}) => { $weight: 10 } greek) | greek" explainscore withscores
Error: No support for nested multi bulk replies with depth > 7
```

My index looks like this:

```
FT.CREATE books-idx ON HASH PREFIX 1 ru203:book:details: SCHEMA isbn TAG SORTABLE title TEXT WEIGHT 2.0 SORTABLE subtitle TEXT SORTABLE thumbnail TAG NOINDEX description TEXT SORTABLE published_year NUMERIC SORTABLE average_rating NUMERIC SORTABLE authors TEXT SORTABLE categories TAG SEPARATOR ";" author_ids TAG SEPARATOR ";"
```

Example data:

```
❯ redis-cli
127.0.0.1:6379> hgetall ru203:book:details:9780385495547
 1) "categories"
 2) "History"
 3) "author_ids"
 4) "1262"
 5) "description"
 6) "Examines the remarkable legacy of the ancient Greeks, from the origins of Greek culture to the development of Western literature, drama, poetry, and philosophy to the Greek influence on human science, mathematics, and logic."
 7) "thumbnail"
 8) "http://books.google.com/books/content?id=sS16vB0pjR0C&printsec=frontcover&img=1&zoom=1&source=gbs_api"
 9) "subtitle"
10) "Why the Greeks Matter"
11) "average_rating"
12) "3.77"
13) "title"
14) "Sailing the Wine-Dark Sea"
15) "isbn"
16) "9780385495547"
17) "authors"
18) "Thomas Cahill"
19) "published_year"
20) "2004"
```

Redis version:
```
❯ redis-cli --version
redis-cli 6.0.9
 
❯ redis-server --version
Redis server v=6.0.9 sha=00000000:0 malloc=libc bits=64 build=26c3229b35eb3beb
```

Docker image:
```
redislabs/redisearch    latest              2ad01ad2b2a6        5 weeks ago         145MB
```

**Comment by ashtul** [comment]
This is an issue within `hiredis`.

**Comment by ashtul** [comment]
Fixed in #1786 
New Redis releases (this is yet to be released as part of a stable version) can output with a depth of more than 7. For older versions, the output will print up to the depth limit of `7` (instead of crashing).
